piptools/resolver.py

```diff
@@ -203,24 +228,32 @@ def _resolve_one_round(self):
 
         # NOTE: We need to compare RequirementSummary objects, since
         # InstallRequirement does not define equality
-        diff = {RequirementSummary(t) for t in theirs} - {RequirementSummary(t) for t in self.their_constraints}
-        removed = ({RequirementSummary(t) for t in self.their_constraints} -
-                   {RequirementSummary(t) for t in theirs})
-        unsafe = ({RequirementSummary(t) for t in unsafe_constraints} -
-                  {RequirementSummary(t) for t in self.unsafe_constraints})
+        diff = {RequirementSummary(t) for t in theirs} - {
+            RequirementSummary(t) for t in self.their_constraints
+        }
+        removed = {RequirementSummary(t) for t in self.their_constraints} - {
+            RequirementSummary(t) for t in theirs
+        }
+        unsafe = {RequirementSummary(t) for t in unsafe_constraints} - {
+            RequirementSummary(t) for t in self.unsafe_constraints
+        }
 
         has_changed = len(diff) > 0 or len(removed) > 0 or len(unsafe) > 0
         if has_changed:
-            log.debug('')
-            log.debug('New dependencies found in this round:')
+            log.debug("")
+            log.debug("New dependencies found in this round:")
             for new_dependency in sorted(diff, key=lambda req: key_from_req(req.req)):
-                log.debug('  adding {}'.format(new_dependency))
-            log.debug('Removed dependencies in this round:')
-            for removed_dependency in sorted(removed, key=lambda req: key_from_req(req.req)):
-                log.debug('  removing {}'.format(removed_dependency))
-            log.debug('Unsafe dependencies in this round:')
-            for unsafe_dependency in sorted(unsafe, key=lambda req: key_from_req(req.req)):
-                log.debug('  remembering unsafe {}'.format(unsafe_dependency))
+                log.debug("  adding {}".format(new_dependency))
+            log.debug("Removed dependencies in this round:")
+            for removed_dependency in sorted(
+                removed, key=lambda req: key_from_req(req.req)
+            ):
+                log.debug("  removing {}".format(removed_dependency))
+            log.debug("Unsafe dependencies in this round:")
+            for unsafe_dependency in sorted(
+                unsafe, key=lambda req: key_from_req(req.req)
+            ):
+                log.debug("  remembering unsafe {}".format(unsafe_dependency))
```

Comment:
Is this block really more readable than the way it was before?

Reply:
Black is not about opinions.. ;)

I find this change OK (apart from the clutter that changing quotes brings here), but black can certainly make things less readable (https://github.com/ambv/black/issues/601).

Reply:
The three lines aren't that much worse on their own - I'm referring to the entire block of code. Reading the previous section, the vertical flow visually reads better, while the new code doesn't read as well visually. 

> Black is not about opinions.. ;)

Isn't this exactly the point? Black has opinions so you don't have to?

Reply:
> Isn't this exactly the point? Black has opinions so you don't have to?

Yeah, that's what I've meant - "Black is not about (your) opinions".

I agree that the flow is worse now.

We could use "# fmt: off" / "# fmt: on" markers here.

Reply:
IMO, this block looks okay (i've seen much worse). Anyways, PRs are welcome to fix "ugly" formattings.